src/lib.rs

```diff
@@ -0,0 +1,1040 @@
+use lazy_static::lazy_static;
+
+extern crate wapc_guest as guest;
+use guest::prelude::*;
+
+use anyhow::{anyhow, Result};
+use k8s_openapi::api::core::v1 as apicore;
+
+extern crate kubewarden_policy_sdk as kubewarden;
+use kubewarden::{logging, protocol_version_guest, request::ValidationRequest, validate_settings};
+
+mod settings;
+use settings::Settings;
+
+use slog::{info, o, Logger};
+
+lazy_static! {
+    static ref LOG_DRAIN: Logger = Logger::root(
+        logging::KubewardenDrain::new(),
+        o!("policy" => "sample-policy")
+    );
+}
+
+#[no_mangle]
+pub extern "C" fn wapc_init() {
+    register_function("validate", validate);
+    register_function("validate_settings", validate_settings::<Settings>);
+    register_function("protocol_version", protocol_version_guest);
+}
+
+#[allow(dead_code)]
+#[derive(Debug, PartialEq)]
+enum PolicyResponse {
+    Accept,
+    Reject(String),
+}
+
+/* Trait to abstract apicore::Container and apicore::EphemeralContainer
+ * This is used to allow using the same generic function to validate the
+ * security context from both structs.
+ */
+trait HasSecurityContext {
```

Comment:
I don't like the name. To me "has" means something returning a boolean. Why don't you call the trait `GetSecurityContext` and then rename the function to be just `security_context`?